resolver/src/main/java/io/netty/resolver/InetNameResolver.java

```diff
@@ -40,32 +37,15 @@
      */
     protected InetNameResolver(EventExecutor executor) {
         super(executor);
-        loopbackAddress = SocketUtils.loopbackAddress();
     }
 
     /**
+     * @deprecated Use {@link SocketUtils#loopbackAddress()}.
      * Returns the {@link InetAddress} for loopback.
      */
+    @Deprecated
     protected InetAddress loopbackAddress() {
-        return loopbackAddress;
-    }
-
-    @Override
-    public Future<InetAddress> resolve(String inetHost, Promise<InetAddress> promise) {
```

Comment:
@Scottmitch didn't we say we want to have the logic in `InetNameResolver` to make it easier to have a consistent behaviour when write a new implementation ?